app/src/main/java/communi/dog/aplicatiion/DB.java

```diff
@@ -167,11 +167,18 @@ public void addUser(String userId, String userEmail, String userPassword, String
         this.usersRef.child(userId).setValue(newUser);
     }
 
-    public boolean isValidUserPassword(String userId, String userPassword) {
+    public String isValidUserPassword(String userId, String userPassword) {
         if (users.containsKey(userId)) {
-            return users.get(userId).getPassword().equals(userPassword);
+            if(users.get(userId).getPassword().equals(userPassword)){
+                return "ok";
+            }
+            else{
+                return "incorrect password";
+            }
+        }
+        else{
+            return "id is unknown";
         }
-        return false;
```

Comment:
I think that a better practice will be to change the return value to an Enum with three options:
* correct userId and password
* unknown userId
* incorrect password
in my opinion it is a better practice than returning a string because who ever uses this function can not guess what are the expected string of each case.
If you don't agree with me tell me and we figure out the best way to solve it :)